nautobot/ipam/factory.py

```diff
@@ -304,3 +361,184 @@ def _adjust_kwargs(cls, **kwargs):
             kwargs["site"] = kwargs["group"].site
 
         return kwargs
+
+
+class VLANGetOrCreateFactory(VLANFactory):
+    class Meta:
+        django_get_or_create = ("location", "site", "tenant")
+
+
+class VRFGetOrCreateFactory(VRFFactory):
+    class Meta:
+        django_get_or_create = ("tenant",)
+
+
+class PrefixFactory(PrimaryModelFactory):
+    """
+    Create random prefixes and 50% of the time generate child prefixes and ip addresses within the prefix IP space.
+    Containers will create child prefixes while prefixes that are not containers will create ip addresses in the
+    prefix's address space. Defaults to creating 0-4 children. Set children__max_count to an integer when calling
+    the factory creation methods (`create()`, `create_batch()`, etc) to override the maximum number of children
+    generated.
+
+    Examples:
+    PrefixFactory.create_batch(20) # create 20 prefixes, approximately half will generate 0-4 children
+    AggregateFactory.create_batch(children__max_count=0) # create 20 prefixes with no children
+    """
+
+    class Meta:
+        model = Prefix
+
+    class Params:
+        has_description = factory.Faker("pybool")
+        has_location = factory.Faker("pybool")
+        has_role = factory.Faker("pybool")
+        has_site = factory.Faker("pybool")
+        has_tenant = factory.Faker("pybool")
+        has_vlan = factory.Faker("pybool")
+        has_vrf = factory.Faker("pybool")
+        is_container = factory.Faker("pybool")
+        is_ipv6 = factory.Faker("pybool")
```

Comment:
Did we identify any tradeoff between the `Meta.exclude` approach versus the `Params` approach for these flags? We should settle on one and be consistent about it.

Reply:
I don't see any functional difference. Neither of them are available from a `post_generation` decorated function and both of them are available using the `factory.SelfAttribute` class from within `Params` and factory attributes.